### PR TITLE
Getting started instructions

### PR DESCRIPTION
**Pull request by vitalikdanchenko**
https://github.com/talkable/talkable-ruby/blob/docs/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,110 +19,132 @@ gem "talkable"
...
-  c.api_token = Rails.env.production? ? "1235" : "6789" 
-  # required
-  c.server = 'http://invite.site.com' # fetched from site settings automatically using generator
-  # optional
-  c.js_integration_library = 'http://d2jj/integration/hello.js'  # default
+Talkable.configure do |config|
+  # site slug takes form ENV["TALKABLE_SITE_SLUG"]
+  config.site_slug  = "spree-example"
+
+  # api key takes from ENV["TALKABLE_API_KEY"]
```
Comment (Tonkonozhenko) [inline README.md]: `api key is taken from ENV["TALKABLE_API_KEY"]` or `takes api key from ENV["TALKABLE_API_KEY"]`

Comment (vitalikdanchenko) [reply README.md]: oh, finally I see this grammar mistake, thanks

File: README.md [hunk 2]
```diff
@@ -17,110 +19,132 @@ gem "talkable"
...
-
-create config/initializers/talkable.rb
-update app/controllers/application_controller.rb
-
-create app/controllers/talkable_invite.rb
-create app/views/talkable_invite/show.html.erb
-update app/layouts/application.html.erb # floating widget install
-update app/layouts/_talkable_floating_widget.html.erb
-update config/routes.rb
+Your Talkable site slug: spree-example
```
Comment (bogdan) [inline README.md:48]: Split the command from output so that it is easier to know what to copypaste.

File: README.md (hunk 1 above)
Comment (bogdan) [reply README.md]: Are you sure that we want `ENV["TALKABLE_API_KEY"]` by default? I would consider it inconvinient a variant in current example.

File: README.md [hunk 3]
```diff
@@ -17,110 +19,132 @@ gem "talkable"
 
 ``` sh
 rails generate talkable:install
```
Comment (bogdan) [inline README.md]: You should add to Gemfile first.

File: README.md [hunk 4]
```diff
@@ -129,32 +153,21 @@ User facing GEM API
 
 Generated code:
```
Comment (bogdan) [inline README.md]: I don't think this section is now required.

**Comment by bogdan** [comment]
Make a paragraph regarding post-purchase integration so that it is clear that it is not automated.

File: README.md [hunk 5]
```diff
@@ -129,32 +153,21 @@ User facing GEM API
...
-## integration.js extension
-
-`integration.js` additions. Suppose to be never used directly if using talkable gem
-
-``` js
-talkable.showOffer(offer.show_url)
-```
-
-
 ## Self-Serve UI
```
Comment (bogdan) [inline README.md]: This section needs to be removed.

File: README.md (hunk 1 above)
Comment (vitalikdanchenko) [reply README.md]: it's like an option if you don't want to show your real api key in code

**Comment by vitalikdanchenko** [comment]
@bogdan there are example of it https://github.com/talkable/talkable-ruby/blob/docs/README.md#referral-campaign

**Comment by bogdan** [comment]
@vitalikdanchenko but it needs to be a part of installation

**Comment by bogdan** [comment]
I would like it to be labeled: `Integrate Conversion Points`

File: README.md [hunk 6]
```diff
@@ -17,228 +19,163 @@ gem "talkable"
...
-  c.api_token = Rails.env.production? ? "1235" : "6789" 
-  # required
-  c.server = 'http://invite.site.com' # fetched from site settings automatically using generator
-  # optional
-  c.js_integration_library = 'http://d2jj/integration/hello.js'  # default
-end
-```
-
+Talkable.configure do |config|
+  # site slug takes form ENV["TALKABLE_SITE_SLUG"]
```
Comment (zhuravel) [inline README.md]: takes from → is taken from

File: README.md [hunk 7]
```diff
@@ -17,228 +19,163 @@ gem "talkable"
...
+Talkable::API::Reward.find(visitor_uuid: '8fdf75ac-92b4-479d-9974-2f9c64eb2e09')
+Talkable::API::Person.find(email)
+Talkable::API::Person.update(email, unsubscribed: true)
+Talkable::API::Referral.update(order_number, Talkable::API::Referral::APPROVED)
 ```
 
+## Referral Campaign
 
-## Self-Serve UI
+Registering an origins
```
Comment (zhuravel) [inline README.md]: an origins → an origin

File: README.md [hunk 8]
```diff
@@ -17,228 +19,163 @@ gem "talkable"
...
-    message: 'Click #{offer.claim_links.twitter} and get {{friend_incentive.description}} off on the merchant.com'
-  },
-
+origin = Talkable.register_purchase(
+  email: 'customer@example.com',
+  order_number: 'N1234567',
+  subtotal: 123.45,
+  coupon_code: 'SALE10',
+  traffic_source: 'checkout',
+  campaign_tags: 'post-purchase',
```
Comment (zhuravel) [inline README.md]: Should `campaign_tags` also be an array, like `sharing_channels` ?

File: README.md [hunk 9]
```diff
@@ -17,228 +19,163 @@ gem "talkable"
...
+  campaign_tags: 'invite',
 )
 
-
-offer.configure(
-  twitter: {
-    message: 'Click #{offer.claim_links.twitter} and get {{friend_incentive.description}} off on the merchant.com'
-  },
-
+origin = Talkable.register_purchase(
```
Comment (bogdan) [inline README.md]: It should be a part of installation or at least listed as required things to do.

File: README.md (hunk 8 above)
Comment (vitalikdanchenko) [reply README.md]: it could be both way

**Comment by Tonkonozhenko** [comment]
Looks great!

File: README.md [hunk 10]
```diff
@@ -17,228 +29,178 @@ gem "talkable"
 
 ``` sh
 rails generate talkable:install
```
Comment (bogdan) [inline README.md:45]: Add a note that this is a Rails installation.  We should communicate how manual installation is done.

File: README.md [hunk 11]
```diff
@@ -5,7 +5,19 @@ Talkable Ruby Gem to make your own referral program in Sinatra or Rails applicat
...
+
+Gem requires:
+ - `Ruby` version since `1.9.3`
+ - `Rack` version since `1.5.2`
+
+Gem supports:
+ - `Ruby on Rails` since `4.0.0`
+ - `Sinatra` since `1.4.0`
 
 ## Intallation
```
Comment (bogdan) [inline README.md:22]: Let's describe a kind of plan here:
1. Generate the configuration file (*)
2. Middleware installation (*)
3. Conversion Points integration
   1. Post-Purchase integration
   2. Other conversion points integration 
4. Floating widget integration (*)
5. /invite integration (*)
- \- Automated in Ruby On Rails using a generator

Comment (Tonkonozhenko) [reply README.md:22]: I think easier way (using generator) must be first.

File: README.md [hunk 12]
```diff
@@ -17,228 +29,178 @@ gem "talkable"
...
-talkable.showOffer(offer.show_url)
+```erb
+<div id="talkable-inline-offer-container"></div>
+<%= render 'shared/talkable_offer', offer: @offer, options: {iframe: {container: 'talkable-inline-offer-container'}} %>
 ```
 
+## Registering a purchase
+
+Registering a purchase has to be implemented manually based on your platform.
+> It's highly recommended to have submitted purchases for closing a referral loop.
```
Comment (bogdan) [inline README.md]: why to say that? I think this is required but not recommended.

File: README.md [hunk 13]
```diff
@@ -17,228 +29,178 @@ gem "talkable"
...
+Talkable::API::Offer.find(short_url_code)
+Talkable::API::Share.create(short_url_code, Talkable::API::Share::CHANNEL_SMS)
+Talkable::API::Reward.find(visitor_uuid: '8fdf75ac-92b4-479d-9974-2f9c64eb2e09')
+Talkable::API::Person.find(email)
+Talkable::API::Person.update(email, unsubscribed: true)
+Talkable::API::Referral.update(order_number, Talkable::API::Referral::APPROVED)
 ```
 
-
 ## TODO
```
Comment (bogdan) [inline README.md]: Let's only leave undone items here or even remove it completely.

**Review by bogdan — APPROVED** [review]

**Pull request merged**